src/sentry/middleware/integrations/parsers/slack.py

```diff
@@ -82,19 +92,126 @@ class SlackRequestParser(BaseRequestParser):
     See: `src/sentry/integrations/slack/views`
     """
 
+    def build_loading_modal(self, external_id: str, title: str):
+        return {
+            "type": "modal",
+            "external_id": external_id,
+            "title": {"type": "plain_text", "text": title},
+            "blocks": [
+                {
+                    "type": "section",
+                    "block_id": "loading_block",
+                    "text": {"type": "mrkdwn", "text": "Loading..."},
+                }
+            ],
+        }
+
+    def parse_slack_payload(self, request) -> str | None:
+        try:
+            decoded_body = parse_qs(request.body.decode(encoding="utf-8"))
+            payload_list = decoded_body.get("payload")
+
+            if not payload_list or not isinstance(payload_list, list) or len(payload_list) != 1:
+                raise ValueError(
+                    "Error parsing Slack payload: 'payload' not found or not a list of length 1"
+                )
+
+            payload = json.loads(payload_list[0])
+
+            # Extract action_ts from the payload
+            # we need to grab the action_ts to use as the external_id for the loading modal
+            # https://api.slack.com/reference/interaction-payloads/block-actions
+            actions = payload.get("actions", None)
+            if actions and isinstance(actions, list) and len(actions) == 1:
+                (action,) = actions  # we only expect one action in the list
+                action_ts = action.get("action_ts")
+                if action_ts is None:
+                    raise ValueError(
+                        "Error parsing Slack payload: 'action_ts' not found in 'actions'"
+                    )
+                return payload, action_ts
+            else:
+                raise ValueError(
+                    "Error parsing Slack payload: 'actions' not found or not a list of length 1"
+                )
```

Comment:
It might be good to differentiate these two cases with separate messages when they occur.